src/fuse_reduce.cpp

```diff
@@ -221,11 +224,19 @@ struct find_pointwise_reduce
         auto reduce = r.result;
         auto input  = r.instructions["pointwise"];
 
+        // A broadcast with output shape's rank not matching input wouldn't work.
+        // TODO:  can the same error happen in find_reduce_reduce or find_reduce_pointwise?
+        if(contains(r.instructions, "broadcast"))
+        {
+            if(r.instructions["broadcast"]->get_shape().ndim() != input->get_shape().ndim() or
```

Comment:
Adding @CharlieL7  This is still a draft PR
I just changed this, but I still don't think the condition I set is quite right.  I made a matcher predicate based on the equation at [](https://github.com/ROCm/AMDMIGraphX/blob/4a4c754e0b0e92c5dac729dbccfe1fc0a300c750/src/common.cpp#L51) but in fact dimensions of `{1}` and `{80, 204, 204}` satisfy the condition and I had to throw in a check that the ranks are the same.  What's a more proper condition that will filter out these dimensions?  Why shouldn't this fusion work with different ranks and a multibroadcast?

Reply:
> Why shouldn't this fusion work with different ranks and a multibroadcast?

This is trying to fuse the broadcast over the reduction dimensions, and its only valid for that case. Its not trivial to do in the matcher because you need to know the reduction axes, so I dont think you should solve that in this PR.

Instead in the matcher just check that the rank matches for the input and output which can catch many cases of misuse(not all but it will fix the issue in `speech_transformer`). Thats because the output of a reduction is the same rank as the input so when we broadcast the reduction it will have the same rank.

Reply:
Changed this to not even look at the dimensions; just makes sure the number of dimensions (rank) is the same--otherwise we don't do a fusion here--and if they are the same, assumes the original model had compatible dimensions.